src/strings/nfa-noodlification.cc

```diff
@@ -378,18 +212,10 @@ std::vector<seg_nfa::NoodleWithEpsilonsCounter> seg_nfa::noodlify_for_equation(
     if (product_pres_eps_trans.is_lang_empty()) {
         return {};
     }
-    if (utils::haskey(params, "reduce")) {
-        const std::string& reduce_value = params.at("reduce");
-        if (reduce_value == "forward" || reduce_value == "bidirectional") {
-            product_pres_eps_trans = reduce(product_pres_eps_trans);
-        }
-        if (reduce_value == "backward" || reduce_value == "bidirectional") {
-            product_pres_eps_trans = revert(product_pres_eps_trans);
-            product_pres_eps_trans = reduce(product_pres_eps_trans);
-            product_pres_eps_trans = revert(product_pres_eps_trans);
-        }
+    if (reduce_intersection) {
+        product_pres_eps_trans = reduce(product_pres_eps_trans, nullptr, red_alg);
     }
-    return noodlify_mult_eps(product_pres_eps_trans, { EPSILON, EPSILON-1 }, include_empty);
+    return noodlify_mult_eps(product_pres_eps_trans, { EPSILON, EPSILON-1 }, red_alg);
```

Comment:
It took me a few seconds to decipher `red_alg`. Could we just use the full name?

Reply:
This PR is right now for experimental purposes, so this is very low effort thing.

Reply:
Sure, I was just quickly scrolling through.